src/preferences/node/preference-server-module.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { ContainerModule, } from 'inversify';
+import { ConnectionHandler } from '../../messaging/common';
+import { JsonRpcProxyFactory } from '../../messaging/common/proxy-factory';
+import { IPreferenceServer, IPreferenceClient } from '../common/preference-protocol'
+import { DefaultPreferenceServer } from '../node/default-preference-server'
+import { getRootDir } from '../../filesystem/node/filesystem-server-module'
+import { CompoundPreferenceServer } from '../common/compound-preference-server'
+import { JsonPreferenceServer, PreferencePath } from './json-preference-server'
+import { FileUri } from "../../application/node/file-uri";
+
+import * as path from 'path';
+
+export const preferenceServerModule = new ContainerModule(bind => {
+    const WorkspacePreferenceServer = Symbol('WorkspacePreferenceServer');
+    const UserPreferenceServer = Symbol('UserPreferenceServer');
+
+    bind(DefaultPreferenceServer).toSelf().inSingletonScope();
+
+    bind(WorkspacePreferenceServer).to(JsonPreferenceServer).inSingletonScope();
```

Comment:
As mentioned above `WorkspacePreferenceServer` cannot be a singleton anymore because there can be several clients connected, each of them with the own workspace.

You should try to do something like:
```ts
bind(JsonPreferenceServer).toSelf();
bind(WorkspacePreferenceServer).toDynamicValue(ctx => {
     const workspaceServer = ctx.container.get(WorkspaceServer);
     const preferencePath = workspaceServer.getRoot().then(root => 
          root.resolve(path.resolve('.theia', 'prefs.json')
     );
     const child = ctx.container.child();
     child.bind(PreferencePath).toConstantValue(preferencePath);
     return child.get(JsonPreferenceServer);
});
```

and simlar for  `UserPreferenceServer`